components/appmgt/org.wso2.carbon.appmgt.impl/src/main/java/org/wso2/carbon/appmgt/impl/dao/AppMDAO.java

```diff
@@ -7034,46 +7043,45 @@ public static JSONArray getPolicyGroupXACMLPoliciesByApplication(String uuid) {
 				objPartial.put("THROTTLING_TIER", rs.getString("THROTTLING_TIER"));
 				objPartial.put("USER_ROLES", rs.getString("USER_ROLES"));
 				objPartial.put("URL_ALLOW_ANONYMOUS", rs.getBoolean("URL_ALLOW_ANONYMOUS"));
-				objPartial.put("POLICY_PARTIAL_NAME", getPolicyGroupWisePolicyPartials(rs.getInt("POLICY_GRP_ID"), conn));
+				objPartial.put("POLICY_PARTIAL_NAME",
+						getPolicyGroupWisePolicyPartials(rs.getInt("POLICY_GRP_ID"), conn));
 				objPartialArr.add(objPartial);
 			}
 		} catch (SQLException e) {
-			/*
-			only logging the exception because this function is
-			get directly called by the jaggery pages only to view the data so throwing the exception outside can
-			disturb the users activities
-			*/
-			log.error("SQL Error while executing the query to get policies under each policy group mapped with the application : " + query + " : (Application UUID:" +
-					uuid + ") : " + e.getMessage(), e);
-			objPartialArr = null;
-		} catch (AppManagementException e) {
-			log.error("Error while fetching policy partial details : " + e.getMessage(), e);
-			objPartialArr = null;
+			handleException("SQL Error while executing the query to get policies under each policy group mapped " +
+					"with the application : " + query + " : (Application UUID:" + AppUUID + ") : " + e.getMessage(), e);
 		} finally {
 			APIMgtDBUtil.closeAllConnections(ps, conn, rs);
 		}
 		return objPartialArr;
 	}
 
-
 	/**
-	 * returns the comma separated list of policy partial names to display in the view assets
-	 * @param policyGroupId : policy group id
-	 * @param conn : sql connection
+	 * returns the comma separated list of policy partial names to display in
+	 * the view assets
+	 *
+	 * @param policyGroupId
+	 *            : policy group id
+	 * @param conn
+	 *            : sql connection
 	 * @return comma separated list of policy partial names
-	 * @throws AppManagementException on error
+	 * @throws AppManagementException
+	 *             on error
 	 */
-	public static String getPolicyGroupWisePolicyPartials(Integer policyGroupId, Connection conn) throws AppManagementException {
+	public static String getPolicyGroupWisePolicyPartials(Integer policyGroupId, Connection conn)
+			throws SQLException {
```

Comment:
This is a private method. (Access modifier has been changed later as 'private' with a subsequent commit)